test/extractor/AddElementRuleExtractor.test.ts

```diff
@@ -60,14 +71,81 @@ describe('AddElementRuleExtractor', () => {
         isReference: true
       });
       expect(elementRule).toEqual<ExportableAddElementRule>(expectedRule);
+      expect(codeableReferenceRule).toBeUndefined();
       expect(element.processedPaths).toEqual(
         expect.arrayContaining(['min', 'max', 'short', 'type[0].code', 'type[0].targetProfile[0]'])
       );
     });
 
+    it('should extract an add element rule and an only rule for an element with a CodeableReference type', () => {
+      const element = ProcessableElementDefinition.fromJSON(looseSD.differential.element[15]);
+      const [elementRule, codeableReferenceRule] = AddElementRuleExtractor.process(
+        element,
+        looseSD
+      );
+      const expectedElement = new ExportableAddElementRule('melon');
+      expectedElement.min = 0;
+      expectedElement.max = '*';
+      expectedElement.short = 'Large melon';
+      expectedElement.definition = 'There are so many kinds of melon';
+      expectedElement.types.push({ type: 'CodeableReference' });
+      const expectedOnly = new ExportableOnlyRule('melon');
+      expectedOnly.types.push({
+        type: 'http://hl7.org/fhir/StructureDefinition/Substance',
+        isReference: true
+      });
+      expect(elementRule).toEqual<ExportableAddElementRule>(expectedElement);
+      expect(codeableReferenceRule).toEqual<ExportableOnlyRule>(expectedOnly);
+      expect(element.processedPaths).toEqual(
+        expect.arrayContaining([
+          'min',
+          'max',
+          'short',
+          'definition',
+          'type[0].code',
+          'type[0].targetProfile[0]'
+        ])
+      );
+    });
+
+    it('should extract an add element rule and an only rule for a choice element that includes a CodeableReference type', () => {
+      const element = ProcessableElementDefinition.fromJSON(looseSD.differential.element[16]);
+      const [elementRule, codeableReferenceRule] = AddElementRuleExtractor.process(
+        element,
+        looseSD
+      );
+      const expectedElement = new ExportableAddElementRule('cheese[x]');
+      expectedElement.min = 0;
+      expectedElement.max = '*';
+      expectedElement.short = 'Cheese is versatile';
+      expectedElement.definition = 'You can put cheese on a lot of things. Sometimes you melt it.';
+      expectedElement.types.push({ type: 'string' }, { type: 'CodeableReference' });
+      const expectedOnly = new ExportableOnlyRule('cheeseCodeableReference');
```

Comment:
Oh... I hadn't thought of this scenario, but... this is actually a problem. The only way to apply the reference to the `CodeableReference` is via a type slice path, but... slices aren't allowed in logical models! SUSHI doesn't yell, but the IG Publisher does:
> Rule sdf-1: 'Element paths must be unique unless the structure is a constraint' Failed

So, creating a type slice here is not a viable option. Although we _could_ use a caret rule assignment, that's awful -- and we can't expect our users to do that. We may have to introduce `CodeableReference(A or B)` after all... I'll talk to Mark.